File: July-LeetCoding-Challenge/13-Same-Tree/Same-Tree.cpp
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if (p == NULL and q == NULL) return true;
        if (p == NULL and q) return false;
        if (p and q == NULL) return false;
        if (p->val != q->val) return false;
        return isSameTree(p->left, q->left) and isSameTree(p->right, q->right);
    }
};

File: July-LeetCoding-Challenge/13-Same-Tree/Same-Tree.kt
package not_categorized_easy

class SameTreeKotlin100 {
    fun isSameTree(p: TreeNode?, q: TreeNode?): Boolean = when {
        p == null && q == null -> true
        p == null || q == null || p.`val` != q.`val` -> false
        else -> isSameTree(p.left, q.left) &&
                isSameTree(p.right, q.right)
    }

    class TreeNode(var `val`: Int) {
        var left: TreeNode? = null
        var right: TreeNode? = null
    }
}